src/tests/roc_audio/test_watchdog.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2017 Mikhail Baranov
+ * Copyright (c) 2017 Victor Gaydov
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <CppUTest/TestHarness.h>
+
+#include "roc_audio/watchdog.h"
+#include "roc_core/buffer_pool.h"
+#include "roc_core/heap_allocator.h"
+
+namespace roc {
+namespace audio {
+
+namespace {
+
+enum {
+    MaxBufSize = 5000,
+
+    SamplesPerPacket = 200,
+    NumCh = 2,
+
+    Timeout = 20
+};
+
+core::HeapAllocator allocator;
+core::BufferPool<sample_t> sample_buffer_pool(allocator, MaxBufSize, 1);
+
+class TestFrameReader : public IReader, public core::NonCopyable<> {
+public:
+    void read(Frame& frame) {
+        memset(frame.samples().data(), 0, frame.samples().size() * sizeof(sample_t));
+    }
+};
+
+TestFrameReader test_reader;
+
+} // namespace
+
+TEST_GROUP(watchdog) {
```

Comment:
We should also cover the case when the first timestamp passed to `update()` is larger than `Timeout` (covered by this [bit](https://github.com/dshil/roc/blob/715debb788f1413fbe4503c6c353f23354527fff/src/modules/roc_audio/watchdog.cpp#L45))